### DISCUSSION TITLE
Polytope from 2D screen points

### DISCUSSION THREAD

**Discussion by loopless** [openscenegraph/OpenSceneGraph#1332] [Q&A]
I am trying to create a Polytope from a series of points on the screen ( a convex screen polygon) to pass to the PolytopeIntersector.
I have no issues using the simplified screen rectangle constructor to PolytopeIntersector ( as per the osg examples) but using a convex general screen polygon has eluded me.

For example this 'picker' works as expected where the screen coordinates are normalized

	PolytopeIntersector* picker
		= new PolytopeIntersector ( osgUtil::Intersector::PROJECTION, nx1, ny1, nx2, ny2 );
		
I am unable to construct a polytope that works using  nominally the same 4 points (used to create 4 planes)
Looking in the debugger, the planes of the polytope seem close to the ones built in the other simplified constructor
		
	PolytopeIntersector* picker
		= new PolytopeIntersector ( osgUtil::Intersector::PROJECTION, polytope );

**Comment by xarray** [comment]
Have a look into PolytopeIntersector.cpp and you will understand how screen values (xMin, xMax, yMin, yMax) turn into polytope points:
    _polytope.add(osg::Plane(1.0, 0.0, 0.0, -xMin));
    _polytope.add(osg::Plane(-1.0,0.0 ,0.0, xMax));
    _polytope.add(osg::Plane(0.0, 1.0, 0.0,-yMin));
    _polytope.add(osg::Plane(0.0,-1.0,0.0, yMax));
    _polytope.add(osg::Plane(0.0,0.0,1.0, zNear));

To construct your own polytope, consider add planes instead of points, and use correct CoordinateFrame

**Reply by loopless** [reply]
Thanks, I resolved my issue, it was a stupid mistake on my part. I constructed the polytope correctly, it was how I was intersecting the scene graph that was at faiult
